src/server_state.cpp

```diff
@@ -731,6 +734,14 @@ void wsrep::server_state::sst_received(wsrep::client_service& cs,
 
         gtid = server_service_.get_position(cs);
         wsrep::log_info() << "Recovered position from storage: " << gtid;
+        if (gtid.seqno() >= connected_gtid().seqno())
+        {
+            /* Now the node has all the data the cluster has: part in
+             * storage, part in replication event queue. */
+            lock.lock();
```

Comment:
Move locking out of the `if` block, `connected_gtid_` is shared state between threads.